### DISCUSSION TITLE
[Bug]: Example notebook returns an error

### DISCUSSION THREAD

**Discussion by Nomados** [pybop-team/PyBOP#344] [Q&A] [answered]
### Python Version

3.11

### Describe the bug

example for [LG_M50_ECM] does not work
it throws the error
 

### Steps to reproduce the behaviour

just running Identifying the Parameters section

### Relevant log output

```shell
self._parameter_set[key] = "[input]"
    ~~~~~~~~~~~~~~~~~~~^^^^^
TypeError: 'ParameterSet' object does not support item assignment
```

**Comment by Nomados** [comment]
actually most other examples do not work either

**Comment by NicolaCourtier** [comment]
Hi! Thanks for trying out the examples - please read this response to a similar issue (#341) and see if it works for you:
https://github.com/pybop-team/PyBOP/issues/341#issuecomment-2148345750

**Reply by Nomados** [reply]
Thanks for the help!

since I am interested in ECMs I ran **1-single-pulse-circuit-model.ipynb** and **equivalent-cicuit-identification.ipynb**, and both still do not work with Python 3.11, pybamm 24.1 and pybop 24.3. 
As for the first example it still fails
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[42], line 1
----> 1 optim = pybop.PSO(cost, max_unchanged_iterations=55, threshold=1e-6)
      2 x, final_cost = optim.run()
      3 print("Initial parameters:", cost.x0)

TypeError: PSO.__init__() got an unexpected keyword argument 'max_unchanged_iterations'
```


The second one **equivalent-cicuit-identification.ipynb** goes a bit further 

```
TypeError                                 Traceback (most recent call last)
Cell In[20], line 1
----> 1 optim = pybop.CMAES(cost, max_iterations=300)
      2 x, final_cost = optim.run()
      3 print("Initial parameters:", cost.x0)

TypeError: CMAES.__init__() got an unexpected keyword argument 'max_iterations'
```

**Reply by NicolaCourtier** [reply, marked as answer]
The arguments that can be passed to the optimisers has changed since v24.3 (see #255). I think you've got these example notebooks from the `develop` branch located [here](https://github.com/pybop-team/PyBOP/tree/develop/examples/notebooks), whereas you need to use those on the main branch, located [here](https://github.com/pybop-team/PyBOP/tree/main/examples/notebooks).

The `1-single-pulse-circuit-model.ipynb` does not appear in the `main` branch, but the examples that are there, including `equivalent-circuit-identification.ipynb`, should work for you. Note that the `develop` branch corresponds to the upcoming release (v24.5) and `main` to the current release (v24.3).

**Reply by Nomados** [reply]
I downloaded by tag marking the release (24.3) and the examples worked!
Thanks and hope for a new major release soon! Thank you for your support @NicolaCourtier
